src/calendar/CalendarMonth.jsx

```diff
@@ -107,7 +107,7 @@ const CalendarMonth = React.createClass({
     }
 
     return (
-      <option key={year} value={year}>{year}</option>
+      <option key={year} value={year}>{moment(year, 'YYYY').format('YYYY')}</option>
```

Comment:
The year by default is simply a string, so to have it in arabic in the dropdown, we have to create a moment object in `YYYY` format, and format it in `YYYY`, so it'll be affected by the locale..